awpy/visualization/plot.py

```diff
@@ -326,3 +341,382 @@ def plot_nades(
+            # to find correct color
+            cur_color = min(ct_val / (ct_val + t_val), ct_val) * np.array(
+                [0, 1, 0]
+            ) + min(t_val / (ct_val + t_val), t_val) * np.array([1, 0, 0])
+
+            rect = patches.Rectangle(
+                (nw_x, se_y),
+                width,
+                height,
+                linewidth=1,
+                edgecolor=cur_color,
+                facecolor=cur_color,
+                alpha=1.0,
+            )
+            axes.add_patch(rect)
+        else:
+            print("Tile not found in map:", tile)
+
+    # Plot player positions if given
+    if player_data is not None:
+        t_positions, ct_positions = (
+            player_data["t"]["player-locations"],
+            player_data["ct"]["player-locations"],
+        )
+        if len(t_positions) > 0:
+            transformed_x = [
+                position_transform(map_name, loc[0], "x") for loc in t_positions
+            ]
+            transformed_y = [
+                position_transform(map_name, loc[1], "y") for loc in t_positions
+            ]
+            color_arr = ["#5d79ae"] * len(t_positions)
+            axes.scatter(transformed_x, transformed_y, c=color_arr)
+        if len(ct_positions) > 0:
+            transformed_x = [
+                position_transform(map_name, loc[0], "x") for loc in ct_positions
+            ]
+            transformed_y = [
+                position_transform(map_name, loc[1], "y") for loc in ct_positions
+            ]
+            color_arr = ["#de9b35"] * len(ct_positions)
+            axes.scatter(transformed_x, transformed_y, c=color_arr)
+    axes.axis("off")
+
+
+def plot_map_control_snapshot(
+    map_name: str,
+    tile_values: FrameMapControl,
+    player_pos: dict | None = None,
+    given_fig_ax: tuple[plt.Figure, plt.Axes] | tuple[None, None] = (None, None),
+    save_filepath: str = "",
+) -> None:
+    """Visualize map control for given ct/t map control value dictionaries.
+
+    Args:
+        map_name (str): Map used position_transform call
+        tile_values (dict): List with map control values dictionary for both sides
+            Expected format = [CT Dict, T Dict]
+        player_pos (dict): Dictionary of player positions for each team.
+            Expected format same as output of extract_player_positions
+        given_fig_ax: Optional tuple containing figure and ax objects for plotting
+        save_filepath (str) : Filepath to save the figure to file if required
+
+    Returns: Nothing, all plotting is done on ax object
+
+    Raises:
+        ValueError: If map_name is not in awpy.data.NAV
+    """
+    if map_name not in NAV:
+        msg = "Map not found."
+        raise ValueError(msg)
+    if "ct" not in tile_values.keys() or "t" not in tile_values.keys():
+        msg = "Tile values argument not in expected FrameMapControl format"
+        raise ValueError(msg)
+    ct_tiles, t_tiles = tile_values["ct"], tile_values["t"]
```

Comment:
This check is definitly a linter time one. The map one depends if it is private or public.

Reply:
Addressed already on branch in previous commit.